### PR TITLE
feat: remove flatbuffer entry

### LINKED ISSUE #2805: Remove "missing format header" from Kafka write buffer
Since #2806 we add a format header to our Kafka messages. Once this was deployed for a while we should remove the fallback that accepts messages w/o that header.

### PR DESCRIPTION
**Pull request by tustvold**
~Draft as builds on #3044~

Removes the entry flatbuffer from the codebase

Relates to #2724 

Closes #2805

### CODE REVIEW

File: write_buffer/src/kafka.rs [hunk 1]
```diff
@@ -720,10 +719,17 @@ mod tests {
         let adapter = KafkaTestAdapter::new(conn);
         let ctx = adapter.new_context(NonZeroU32::new(1).unwrap()).await;
 
+        let headers = IoxHeaders::new(ContentType::Protobuf, None);
```
Comment (tustvold) [inline write_buffer/src/kafka.rs:722]: This is necessary because the write buffer now errors if the header isn't present

**Review by alamb — COMMENTED** [review]
❤️

**Review by alamb — APPROVED** [review]

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
This may not be mergable as the `check-flattbuffers` check is removed but the repo has it marked as required. Perhaps @e-dard  could remove that requirement

**Comment by alamb** [comment]
💥  what a great way to go into the weekend

**Pull request merged**